pkg/apps/store/store.go

```diff
@@ -147,39 +148,57 @@ func getCR(cr interface{}) (*customResource, error) {
 	return &customResource{namespace: namespace, size: size, imageRegistries: imageRegistries, registryConfiguration: &registryConfiguration}, nil
 }
 
-func generateRc(v types.PublicVersion, config *protoform.Config, kubeclient *kubernetes.Clientset, customResource *customResource, cr interface{}) ([]*components.ReplicationController, error) {
+func generateRc(v types.PublicVersion, config *protoform.Config, kubeclient *kubernetes.Clientset, sizeClient *sizeclientset.Clientset, customResource *customResource, cr interface{}) ([]*components.ReplicationController, error) {
 	rcs := make([]*components.ReplicationController, 0)
 	// Size
-	sizegen, ok := ComponentStore.Size[v.Size]
-	if !ok {
-		return nil, fmt.Errorf("size %s couldn't be found", v.Size)
+
+	if len(customResource.size) == 0 {
+		return nil, errors.New("size couldn't be found")
 	}
 
-	componentSize := sizegen.GetSize(customResource.size)
-	if componentSize == nil {
-		return nil, fmt.Errorf("size %s couldn't be found in %s", customResource.size, v.Size)
+	var componentSize *sizev1.Size
+	var err error
+
+	if config.DryRun {
+		componentSize = size.GetSize(customResource.size)
+		if componentSize == nil {
+			return nil, fmt.Errorf("couldn't find size %s", customResource.size)
+		}
+	} else {
+		componentSize, err = sizeClient.SynopsysV1().Sizes(config.Namespace).Get(strings.ToLower(customResource.size), metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	// RC
 	for k, v := range v.RCs {
-		rcSize, ok := componentSize[k]
-		if !ok {
-			return nil, fmt.Errorf("replication controller %s couldn't be found in size [%s]", k, customResource.size)
-		}
+		rcSize, rcFound := componentSize.Spec.Rc[k]
+
 		rc := &types.ReplicationController{
 			Namespace:  customResource.namespace,
-			Replicas:   rcSize.Replica,
+			Replicas:   1,
 			Containers: map[types.ContainerName]types.Container{},
 		}
 
+		if rcFound {
+			rc.Replicas = rcSize.Replica
+		}
+
 		for containerName, defaultImage := range v.Container {
-			rc.Containers[containerName] = types.Container{
-				Image:  generateImageTag(defaultImage, customResource.imageRegistries, customResource.registryConfiguration),
-				MinMem: componentSize[k].Containers[containerName].MinMem,
-				MaxMem: componentSize[k].Containers[containerName].MaxMem,
-				MinCPU: componentSize[k].Containers[containerName].MinCPU,
-				MaxCPU: componentSize[k].Containers[containerName].MaxCPU,
+			tmpContainer := types.Container{
+				Image: generateImageTag(defaultImage, customResource.imageRegistries, customResource.registryConfiguration),
+			}
+			if rcFound {
+				containerSize, containerSizeFound := rcSize.Containers[string(containerName)]
```

Comment:
The 2 lines can be simplified to if containerSize, containerSizeFound := rcSize.Containers[string(containerName)]; containerSizeFound {
////
}